src/types.rs

```diff
@@ -7,6 +10,24 @@ use rocket::response::Debug;
 // [rocket::response::Debug](https://api.rocket.rs/v0.5-rc/rocket/response/struct.Debug.html) implements Responder to Error
 pub type Result<T, E = Debug<Error>> = std::result::Result<T, E>;
 
+#[derive(Responder, Debug)]
+#[response(status = 400)]
+pub struct ErrorMessage(pub &'static str);
+
+#[derive(Responder, Debug)]
+#[response(status = 400)]
+pub struct ErrorMessageInfo(pub String);
+
+#[derive(Responder, Debug)]
+pub enum ApiError {
```

Comment:
I really like this type :-)

Reply:
Probably we should take a closer look though at http error codes returned by each of then. E.g. DB error should return 500, Invalid Payload 400, Missing Record 404, Missing signature 401 etc